pmlc/ast/eval.cc

```diff
@@ -178,9 +178,13 @@ int64_t Evaluator::evaluate(const DimNode *node) {
         case AffineOp::Div:
           return lhs / rhs;
         case AffineOp::Max:
-          return std::max(lhs, rhs);
+          // Surfacing of error due to not being implemented at builder level
+          throw std::runtime_error("NYI: AffineOp::Max");
```

Comment:
Not sure I understand this one, if you comment out the implementation, then yes it's NYI.

Reply:
Investigating this now because it appears that I misinterpretted what was going on between here and the builder level; there's a similar "NYI" message at the builder level for min and max AffineOps, and I was aiming to surface this to where those ops are added at the construction level. But it looks like things that are AffineOp::Max here don't necessarily correspond to that at the build level, as seen by resnet50_test failing in the merge checks because it contains construction-level AffineOp::Max but no builder-level AffineOp::Max. Taking a second look at this (and min) now.

Reply:
OK, I remember this now. There are 2 ways things that the evaluator is used for. 
1. DimNode
2. PolyNode

For the `DimNode` case, min/max is sensible and is implemented here.
For the `PolyNode` case, it's not clear what the right implementation would be. This is a question for @tzerrell, it might just be invalid in general.

Reply:
Probably min/max of a PolyNode is straight-up invalid; the min (or max) of two (affine) polynomials is not (affine) polynomial itself, but is instead _piecewise_ (affine) polynomial. IIRC this is not something we want anywhere, but also is something that is supported with the full semi-affine generality of MLIR Affine dialect... I could definitely be mis-remembering either part of that though. @jbruestle might know more about what we intend & are able to support here.

Reply:
Regardless, it sounds like (for now at least) keeping this error as-is at the builder level is appropriate as it properly distinguishes the invalid/nyi min/max operations. My attempt at surfacing these errors has been removed.